lib/prometheus/client/registry.rb

```diff
@@ -22,7 +22,7 @@ def register(metric)
         name = metric.name
 
         @mutex.synchronize do
-          if exist?(name.to_sym)
+          if @metrics.key?(name.to_sym)
```

Comment:
I made this change because `Mutex` is not re-entrant. I guess we could use a `Monitor` instead but inlining the `exist?` method actually makes the code slightly more readable in my opinion.